File: src/org/cerion/symcalc/expression/function/core/CompoundExpression.java
package org.cerion.symcalc.expression.function.core;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.function.Function;
import org.cerion.symcalc.expression.function.FunctionExpr;

public class CompoundExpression extends FunctionExpr {

    public CompoundExpression(Expr... e) {
        super(Function.COMPOUND_EXPRESSION, e);
    }

    @Override
    protected Expr evaluate() {
        getAll(); // eval all
        return get(size() - 1);
    }
}

File: src/main/kotlin/function/core/CompoundExpression.kt
package org.cerion.symcalc.function.core

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.function.FunctionExpr

class CompoundExpression(vararg e: Expr) : FunctionExpr(*e) {

    override fun evaluate(): Expr {
        return get(size - 1)
    }

    override fun validate() {
    }
}
